dio/lib/src/transformers/fused_transformer.dart

```diff
@@ -88,9 +89,12 @@ class FusedTransformer extends Transformer {
     if (isJsonContent && decodedResponse != null) {
       // slow path decoder, since there was a custom decoder specified
       return jsonDecode(decodedResponse);
-    } else if (decodedResponse != null) {
+    } else if (customResponseDecoder != null) {
```

Comment:
Is this making a difference compared to the previous? I'm seeing more readability if it's `decodedResponse != null`.

Reply:
Yes, tests failed after I changed the default transformer because of that.

Custom decoders can return null.
Since there are tests that rely on this, there also might be users.